plenario/api/condition_builder.py

```diff
@@ -20,80 +21,78 @@
+    try:
+        return _parse_condition_tree(table, condition_tree, literally)
+    except Exception as ex:
+        raise ValueError('{} caused parse to fail for table {} with args {}'
+                         .format(ex, table, condition_tree))
 
-    field_tokens = field.split("__")
-    column = table.columns.get(field_tokens[0])
 
-    # equality filters based on table columns
-    if len(field_tokens) == 1 and column is not None:
-        return _operator_to_condition(column, 'eq', value)
+def _parse_condition_tree(table, ctree, literally=False):
+    """Parse nested conditions provided as a dict for a single table.
 
-    # comparison filters based on table columns
-    elif len(field_tokens) == 2 and column is not None:
-        op = field_tokens[1]
+    :param table: table object whose columns are being used in the conditions
+    :param ctree: dictionary of conditions created from JSON
 
-        if op in field_ops:
-            return _operator_to_condition(column, op, value)
+    :returns SQLAlchemy conditions for querying the table with"""
 
+    op = ctree['op']
 
-def _operator_to_condition(column, operator, operand):
+    if op == "and":
+        return and_(
+            _parse_condition_tree(table, child, literally)
+            for child in ctree['val']
+        )
+
+    elif op == "or":
+        return or_(
+            _parse_condition_tree(table, child, literally)
+            for child in ctree['val']
+        )
+
+    elif op in field_ops:
+        col = ctree['col']
+        val = ctree['val']
+        try:
+            return _operator_to_condition(
+                getattr(table, 'columns')[col], op, val, literally
+            )
+
+        # Exists for date__time_of_day. Since it doesn't come
+        except KeyError:
+            return getattr(col, field_ops[op])(val)
+
+
+def _operator_to_condition(column, operator, operand, literally=False):
     """Convert an operation into a SQLAlchemy condition. Operators
     are mapped to SQLAlchemy methods with the field_ops dictionary.
 
     :param column: column object from the table to build condition for
     :param operator: string name of the desired operator
     :param operand: some target value or parameter
+    :param literally: return condition as a string literal
 
-    :returns a string SQLAlchemy condition"""
+    :returns: SQLAlchemy condition or string"""
 
     if operator == 'in':
-        return column.in_(operand.split(','))
+        condition = column.in_(operand.split(','))
     elif operator == 'eq':
-        return column == operand
+        condition = column == operand
     else:
-        return getattr(column, field_ops[operator])(operand)
-
-
-# general_filters
-# ===============
-# Filters which apply to all the tables, but whose columns are not directly
-# specified by an argument key. ex: 'obs_date__ge' translates into a condition
-# that needs to be made for a table's 'point_date' column.
-
-general_filters = {
```

Comment:
Since these conversions were handled exclusively by `parse_general`, they are also not needed.